### PR TITLE
draggableGroup features

### PR DESCRIPTION
**Pull request by muthukumarse**
- [ ] slick.grid - changed getHeaderRowColumn, due to not returning column as expected - verified not used anywhere so far
- [ ] added grouping icon through onHeaderCellRendered
- [ ] added GroupToggler to support little icon for expand/collpase
- [ ] added internal setupColumnReorder in order to avoid manipulating slick.grid which is used on other places
- [ ] exposed function to push default Grouping on load through setDroppedGroups
- [ ] updated function clearDroppedGroups
- [ ] added notify to know from impl newly added/removed group

### CODE REVIEW

**Comment by muthukumarse** [comment]
Hey Bro, I have cleanup and pushed the changes. Tested my side. 

Let me know once you have look.

**Comment by ghiscoding** [comment]
@6pac @muthukumarse 
I see 1 line change in the `slick.grid.js` file which could have big impact on peoples code. is that change really necessary and was it a bug that you fixed on that line?

**Comment by muthukumarse** [comment]
@6pac - slick.grid.js changes is required. since its bug. Also checked across nowhere utilised so far.
expectation - its should return the column object, but was returning empty always.

you may quick do quick find too, I don't see anywhere used.

**Comment by ghiscoding** [comment]
Thanks for the answer, I just wanted to make it was really a bug before, since it could have an impact on users because the function `getHeaderRowColumn()` is exposed to the public. 

@6pac 
You have more knowledge on that, so I'll let you continue the conversation.

**Comment by muthukumarse** [comment]
@ghiscoding - I assume that its bug, since nowhere used so far. So mostly we missed out earlier.
Also it should return Column object.

@6pac - let us know if you able to recall the usage of that function

@ghiscoding - oops i missed you on last comment

**Comment by 6pac** [comment]
Just checking that this is the only PR for this change (it includes #205 ?).

Thanks for removing the slick.grid.js mods, I thought it would be relatively easy to keep them in the plugin.   
That remaining line, yes it does appear to be a bug, if it's not workin. Not sure the fix above is what we need, I'll look at it and get back to you.

**Comment by muthukumarse** [comment]
@6pac - yes its same as #205 with clean changes
sure let us know once you go through.

**Comment by 6pac** [comment]
Hi @muthukumarse , I just changed slick.grid.js to be a bit more accommodating of this plugin.
```getHeaderRowColumn()``` is working, but the 'HeaderRow' is a different thing from the 'Header', and it's not configured in this case, hence the empty return values.

I've created a public ```getHeader()``` and ```getHeaderColumn``` that should do the job. Can you please use these instead in your PR, and remove your changes to slick.grid.js.  Then I'll happily action this PR.

File: plugins/slick.draggablegrouping.js [hunk 1]
```diff
@@ -30,10 +30,12 @@
     var _dataView;
     var dropbox;
     var dropboxPlaceholder;
+    var groupToggler;
     var _self = this;
     var _defaults = {
     };
-    
+    var onGroupChanged = new Slick.Event();
```
Comment (ghiscoding) [inline plugins/slick.draggablegrouping.js]: please provide an "unsubscribe" to this event inside the "destroy()" function

**Comment by ghiscoding** [comment]
@muthukumarse 
There's something wrong about your PR, it includes all commits there were made recently. We should only see your changes, not the entire list of changes since you created the PR.

**Comment by muthukumarse** [comment]
thanks @ghiscoding 

I am closing this PR and initiate new one with clean changes with recent changes.

**Comment by ghiscoding** [comment]
Sorry about that but yeah it's probably the best thing to do.

**Pull request closed without merging**